### PR TITLE
Update test subset file

### PR DESCRIPTION
**Pull request by vivbak**
Fixes including: 
- Handles projects that don't have pedigree information, but require participants to be transferred 
- Fixes pedigree export type 
- Handles case where sample might exist already in the test project, but analysis would still need to be transferred 
- Updates API call 
- Creates a new function to get random families that is a little less random. It aims to evenly distribute the families that are chosen. 
For example, if there are 2 families of size 3, 2 families of size 4 and 2 families of size 5, the and the function has been asked to sample 3 families, it will pull 1 family of each size.

### CODE REVIEW

File: scripts/create_test_subset.py [hunk 1]
```diff
@@ -394,6 +452,54 @@ def _print_fam_stats(families: List):
...
+            returned_families.extend(random.sample(fams, n_pull))
+
+    else:
+        # we can't evenly distribute, so we'll just pull randomly
+        returned_families = random.sample(families_within_threshold, families_n)
+
+    return returned_families
+
+
 def _copy_files_in_dict(d, dataset: str, sid_replacement: Optional[Tuple] = None):
```
Comment (vladsavelyev) [inline scripts/create_test_subset.py:503]: Missing return type hint

File: scripts/create_test_subset.py [hunk 2]
```diff
@@ -280,6 +309,35 @@ def transfer_ped(initial_project, target_project, family_ids):
     return external_participant_ids
 
 
+def transfer_participants(initial_project, target_project, participant_ids):
```
Comment (vladsavelyev) [inline scripts/create_test_subset.py:312]: Please add type hints. (I know I'm guilty of sometimes leaving them out too 😬)

File: scripts/create_test_subset.py [hunk 3]
```diff
@@ -105,9 +114,18 @@ def main(
     if families_n is not None:
         fams = fapi.get_families(project)
         all_families = [family['id'] for family in fams]
-        families = random.sample(all_families, families_n)
-        pedigree = fapi.get_pedigree(project=project, internal_family_ids=families)
+        full_pedigree = fapi.get_pedigree(
+            project=project, internal_family_ids=all_families
+        )
+        external_family_ids = _get_random_families(full_pedigree, families_n)
```
Comment (vladsavelyev) [inline scripts/create_test_subset.py:120]: You mentioned you ended up selecting families manually. Perhaps we could use an option to provide family IDs as input to the script?

File: scripts/create_test_subset.py [hunk 4]
```diff
@@ -394,6 +452,54 @@ def _print_fam_stats(families: List):
...
 
 
+def _get_random_families(
+    families: List,
+    families_n: int,
+    include_single_person_families: Optional[bool] = False,
+):
+    """Obtains a subset of families, that are a little less random.
+    By default single-person families are discarded.
+    The function aims to evenly distribute the families chosen by size.
```
Comment (vladsavelyev) [inline scripts/create_test_subset.py:462]: Nice!

File: scripts/create_test_subset.py (hunk 3 above)
Comment (vladsavelyev) [reply scripts/create_test_subset.py:120]: Or with `_get_random_families` the selection ends up better?

**Review by vladsavelyev — APPROVED** [review]
I like the families selection function!

File: scripts/create_test_subset.py (hunk 3 above)
Comment (vivbak) [reply scripts/create_test_subset.py:120]: Yeah so _get_random_families just does what I was doing with that manual selection :)

**Pull request merged**